nullaway/build.gradle

```diff
@@ -33,6 +33,7 @@ dependencies {
     compileOnly deps.build.errorProneCheckApi
     implementation deps.build.checkerDataflow
     implementation deps.build.guava
+    implementation deps.build.jspecify
```

Comment:
@lazaroclapp just wanted to confirm, are we ok shipping NullAway with a dependence on the 0.2.0 version of the jspecify artifact?

Reply:
Makes sense to me. Also, I am hoping JSpecify versions are fairly backward compatible? To the point that if something requires 1.0 and we are still asking for 0.3... well, as long as `@NullMarked` still has the same FQN, we should be good?

Reply:
Yes, I agree that most likely we should be fine in terms of backward compatibility.

Reply:
This is a bit of a digression into something I don't fully understand, but: I had the impression that `getAnnotation(Class)` can be a problem if the class is not on the _classpath_ (as in, the processorpath isn't necessarily sufficient). See, e.g., https://github.com/google/error-prone/commit/99ed9b2dba942907bc14a087c93698718ddab9ab#diff-1e803625f23f1adba74002be68aec5bf7771e6aa453e1813e582ce1f12c89e02R489. But I've always been confused about the details of this. Plus, maybe the problem exists only if the annotation has certain kinds of elements (enum or class?)? So you _might_ benefit from avoiding that API, but I'm not sure.

The workaround in Error Prone looks to be [looking at `getQualifiedName()` on each item of `getRawAttributes()`](https://github.com/google/error-prone/blob/53851ac838af223166f7027068ad407d1223d628/check_api/src/main/java/com/google/errorprone/util/ASTHelpers.java#L864). Or you could use the simple-name approach that you mentioned.

(RE: backward compatibility of FQN: I hope that we maintain it so as to avoid causing problems here. Feel free to tell us that that would be useful to promise :))

Reply:
Thanks, @cpovirk!  That is concerning.  I think we better migrate to matching by FQN using the relevant `ASTHelpers` method, to be safe.